src/Stratis.Bitcoin.Features.Api/Program.cs

```diff
@@ -71,11 +71,30 @@ public class Program
                 })
                 .UseStartup<Startup>();
 
-            IWebHost host = webHostBuilder.Build();
-                
-            host.Start();
+            bool retry = apiSettings.ApiPort == 0;
+            int retryCnt = retry ? 10 : 1;
 
-            return host;
+            while (retryCnt-- >= 0)
```

Comment:
We should not interfere with default ports (I understand searching for port in test methods but not in the node start-up)

Reply:
The behavior is only triggered if a port number of 0 is used.

Reply:
This should not interfere with the default ports as the logic is only triggered by a port number of 0.